### ISSUE TITLE
podman top hpid with cgroup v2 fails with "Error: couldn't find pids group for PID"

### ISSUE THREAD

**Issue by TristanCacqueray** [containers/libpod#4246]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

podman top $cid hpid fails with "Error: couldn't find pids group for PID"

**Steps to reproduce the issue:**

1. podman run --rm --name test fedora sleep inf

2. podman top test hpid

**Describe the results you received:**

```
Error: couldn't find pids group for PID 2562
```

**Describe the results you expected:**

```
HPID
2562
```

**Additional information you deem important (e.g. issue happens only occasionally):**

This doesn't happen with version 1.5.1

**Output of `podman version`:**

```
Version:            1.6.1
RemoteAPI Version:  1
Go Version:         go1.13
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.13
  podman version: 1.6.1
host:
  BuildahVersion: 1.11.2
  CgroupVersion: v2
  Conmon:
    package: conmon-2.0.1-1.fc31.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.1, commit: 5e0eadedda9508810235ab878174dca1183f4013'
  Distribution:
    distribution: fedora
    version: "31"
  MemFree: 14221209600
  MemTotal: 16659103744
  OCIRuntime:
    package: crun-0.10.2-1.fc31.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.10.2
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  SwapFree: 8396992512
  SwapTotal: 8396992512
  arch: amd64
  cpus: 8
  eventlogger: journald
  hostname: localhost.localdomain
  kernel: 5.3.4-300.fc31.x86_64
  os: linux
  rootless: true
  slirp4netns:
    Executable: /usr/bin/slirp4netns
    Package: slirp4netns-0.4.0-20.1.dev.gitbbd6f25.fc31.x86_64
    Version: |-
      slirp4netns version 0.4.0-beta.3+dev
      commit: bbd6f25c70d5db2a1cd3bfb0416a8db99a75ed7e
  uptime: 10m 18.6s
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ConfigFile: /var/home/tristanc/.config/containers/storage.conf
  ContainerStore:
    number: 8
  GraphDriverName: overlay
  GraphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-0.6.4-2.fc31.x86_64
      Version: |-
        fusermount3 version: 3.6.2
        fuse-overlayfs: version 0.6.4
        FUSE library version 3.6.2
        using FUSE kernel interface version 7.29
  GraphRoot: /var/home/tristanc/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 25
  RunRoot: /run/user/1000
  VolumePath: /var/home/tristanc/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.6.1-2.fc31.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Silverblue 31 physical host

**Comment by mheon** [comment]
What was the command line the container was started with? Any chance it was `--pid=host` / rootless / `--cgroups=disable`

**Comment by TristanCacqueray** [comment]
The container was started by a regular user rootless.

**Comment by mheon** [comment]
Rootless doesn't have cgroups, so no PID cgroup, so this looks expected. We should improve documentation and errors for `top` to indicate that we don't expect some parts of it to work as rootless on CGroups v1.

**Comment by TristanCacqueray** [comment]
Podman does find the correct pid, but the command now fails (and output the pid as the last word instead of within new line). How are we supposed to get the host pid of a container with CGroups v2?

**Comment by mheon** [comment]
We can get the PID of the first process in the container (it'll pop up in `inspect`), but without the PID cgroup we can't track anything beyond that first PID.

**Comment by TristanCacqueray** [comment]
IIUC podman does create the PID cgroup:

```
[fedora@localhost ~]$ podman run --detach --rm --name test fedora bash -c "sleep inf & sleep infinity & sleep 3600"
2b035dd16cd88d100b9482e6f9b91a3031a36c95d0de7170916eca2a9b8dac9a

[fedora@localhost ~]$ podman exec test cat /proc/self/cgroup
0::/user.slice/user-1000.slice/user@1000.service/user.slice/libpod-2b035dd16cd88d100b9482e6f9b91a3031a36c95d0de7170916eca2a9b8dac9a.scope

[fedora@localhost ~]$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/user.slice/libpod-2b035dd16cd88d100b9482e6f9b91a3031a36c95d0de7170916eca2a9b8dac9a.scope/cgroup.controllers 
memory pids

[fedora@localhost ~]$ systemd-cgls /user.slice/user-1000.slice/user@1000.service/user.slice/libpod-2b035dd16cd88d100b9482e6f9b91a3031a36c95d0de7170916eca2a9b8dac9a.scope
Control group /user.slice/user-1000.slice/user@1000.service/u>
├─2405 bash -c sleep inf & sleep infinity & sleep 3600
├─2410 sleep inf
├─2411 sleep infinity
└─2412 sleep 3600
```

**Comment by mheon** [comment]
Is this a V2 system? V2 with the systemd cgroup driver will create the PID cgroup.

@vrothberg I know we did some work to get psgo supporting V2. This error looks like it's out of psgo, though, which makes me suspect that we don't have full support yet?

**Comment by TristanCacqueray** [comment]
This is a fedora 31 which seems to use cgroup v2 by default.

**Comment by TristanCacqueray** [comment]
Actually this is already fixed in master, probably by https://github.com/containers/libpod/pull/4197

**Comment by rhatdan** [comment]
Yes we found this issue earlier, so the problem should be fixed when we release podman 1.6.2 or podman 1.7.
